src/averageAges.js

```diff
@@ -37,7 +52,22 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter((person) => person.sex === 'f');
+  let sorted = women;
+
+  if (withChildren === true) {
+    sorted = women.filter((person) => {
+      return people.some(p => p.mother === person.name);
+    });
+  }
+
+  const sumOfYears = sorted.reduce((acc, person) => {
+    const yearsLived = person.died - person.born;
+
+    return acc + yearsLived;
+  }, 0);
+
+  return sumOfYears / sorted.length;
```

Comment:
you are doing the same at least twice (or even 3 times) - move it to a separate function